gherkin/testdata/good/rule_with_tag.feature.ast.ndjson

```diff
@@ -1 +1 @@
+{"gherkinDocument":{"comments":[],"feature":{"children":[{"rule":{"children":[{"scenario":{"description":"","examples":[],"id":"1","keyword":"Scenario","location":{"column":5,"line":7},"name":"Scenario with only a feature tag","steps":[{"id":"0","keyword":"Given ","keywordType":"Context","location":{"column":7,"line":8},"text":"a"}],"tags":[]}}],"description":"  The untagged rule description","id":"2","keyword":"Rule","location":{"column":3,"line":4},"name":"Untagged rule","tags":[]}},{"rule":{"children":[{"scenario":{"description":"","examples":[],"id":"4","keyword":"Scenario","location":{"column":5,"line":14},"name":"Scenario with feature and rule tags","steps":[{"id":"3","keyword":"Given ","keywordType":"Context","location":{"column":7,"line":15},"text":"b"}],"tags":[]}},{"scenario":{"description":"","examples":[],"id":"7","keyword":"Scenario","location":{"column":5,"line":18},"name":"Scenario with feature, rule and scenario tags","steps":[{"id":"5","keyword":"Given ","keywordType":"Context","location":{"column":7,"line":19},"text":"b"}],"tags":[{"id":"6","location":{"column":5,"line":17},"name":"@tag_scenario"}]}},{"scenario":{"description":"","examples":[{"description":"","id":"12","keyword":"Examples","location":{"column":7,"line":26},"name":"","tableBody":[{"cells":[{"location":{"column":11,"line":28},"value":"a"}],"id":"10","location":{"column":9,"line":28}}],"tableHeader":{"cells":[{"location":{"column":11,"line":27},"value":"header"}],"id":"9","location":{"column":9,"line":27}},"tags":[{"id":"11","location":{"column":7,"line":25},"name":"@examples_tag"}]}],"id":"14","keyword":"Scenario Outline","location":{"column":5,"line":22},"name":"Tagged Scenario outline","steps":[{"id":"8","keyword":"Given ","keywordType":"Context","location":{"column":7,"line":23},"text":"b"}],"tags":[{"id":"13","location":{"column":5,"line":21},"name":"@tag_outline"}]}}],"description":"  The tagged rule description","id":"16","keyword":"Rule","location":{"column":3,"line":11},"name":"Tagged rule","tags":[{"id":"15","location":{"column":3,"line":10},"name":"@tag_rule"}]}}],"description":"","keyword":"Feature","language":"en","location":{"column":1,"line":2},"name":"Some tagged rules","tags":[{"id":"17","location":{"column":1,"line":1},"name":"@tag_feature"}]},"uri":"../testdata/good/rule_with_tag.feature"}}
```

Comment:
You mentioned some funkiness about generation, the order of the stuff here seems a bit back to front. But no idea truly 🤷